spec.md

```diff
@@ -0,0 +1,948 @@
+
+The `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` headers indicate in beats the start and end of the medley section of a song.
+Implementations MUST respect the `GAP` value when calculating the medley start and end times.
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> The headers `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` have been replaced by `MEDLEYSTART` and `MEDLEYEND` in version 2.0.0 of this specification.
+
+### 3.16. The `CALCMEDLEY` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       "on" / "off"
+Since:        0.2.0
+```
+
+If `MEDLEYSTART` or `MEDLEYEND` (`MEDLEYSTARTBEAT` or `MEDLEYENDBEAT` for versions before 2.0.0) are not specified,
+the `CALCMEDLEY` header indicates whether an implementation is supposed to determine the start and end of the medley section automatically.
+The value of this header is compared case-insensitively.
+
+> [!CAUTION]
+>
+> The exact semantics of the `CALCMEDLEY` header have not been defined yet.
+
+### 3.17. The `TITLE` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Since:        0.1.0
+```
+
+The `TITLE` header contains the title of the song.
+
+### 3.18. The `ARTIST` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Since:        0.1.0
+```
+
+The `ARTIST` header contains the artist of the song.
+
+### 3.19. The `YEAR` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       4DIGIT
+Since:        0.2.0
+```
+
+The `YEAR` indicates the year in which the song was released.
+The value must be a positive integer.
+
+### 3.20. The `GENRE` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        0.2.0
+```
+
+The `GENRE` defines the genre(s) of the song.
+Individual genre values MUST be compared case-insensitively.
+For consistency, it is usually best to capitalize genres.
+
+> [!CAUTION]
+>
+> Whether genres should be compared case-insensitively or not hasn't been decided yet.
+
+### 3.21. The `LANGUAGE` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        0.2.0
+```
+
+The `LANGUAGE` header indicates the spoken or sung language(s) of a song.
+
+### 3.22. The `EDITION` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        0.2.0
+```
+
+The `EDITION` indicates what edition of games a song belongs to.
+While this header is intended to hold commercially available editions (e.g. SingStar Pop Hits, GuitarHero Live) implementations MUST NOT reject a file based on the value of this header.
+A list of SingStar editions is available [here](https://github.com/bohning/usdb_syncer/wiki/SingStar-Editions).
+For arbitrary keywords see the `TAGS` header.
+
+### 3.23. The `TAGS` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        1.1.0
+```
+
+The `TAGS` allow association of any reasonable keyword with a song.
+Implementations SHOULD compare tags in a case-insensitive manner.
+
+> [!CAUTION]
+>
+> Whether tags should be compared case-insensitively or not hasn't been decided yet.
+
+### 3.24. The `P1`, `P2`, … Headers
+
+```
+Required:     No
```

Comment:
required when P1, P2, Pn is mentioned in the song body.

Reply:
Fixed